curp/src/rpc/connect.rs

```diff
@@ -468,6 +467,119 @@ impl InnerConnectApi for Connect<InnerProtocolClient<Channel>> {
+    }
+
+    /// Update server addresses, the new addresses will override the old ones
+    async fn update_addrs(&self, _addrs: Vec<String>) -> Result<(), tonic::transport::Error> {
+        // bypassed connect never updates its addresses
+        Ok(())
+    }
+
+    /// Send `ProposeRequest`
+    async fn propose(
+        &self,
+        request: ProposeRequest,
+        _timeout: Duration,
+    ) -> Result<tonic::Response<ProposeResponse>, CurpError> {
+        let mut req = tonic::Request::new(request);
+        req.metadata_mut().inject_current();
+        self.server.propose(req).await.map_err(Into::into)
+    }
+
+    /// Send `PublishRequest`
+    async fn publish(
+        &self,
+        request: PublishRequest,
+        _timeout: Duration,
+    ) -> Result<tonic::Response<PublishResponse>, CurpError> {
+        let mut req = tonic::Request::new(request);
+        req.metadata_mut().inject_current();
+        self.server.publish(req).await.map_err(Into::into)
+    }
+
+    /// Send `ProposeRequest`
+    async fn propose_conf_change(
+        &self,
+        request: ProposeConfChangeRequest,
+        _timeout: Duration,
+    ) -> Result<tonic::Response<ProposeConfChangeResponse>, CurpError> {
+        let mut req = tonic::Request::new(request);
+        req.metadata_mut().inject_current();
+        self.server
+            .propose_conf_change(req)
+            .await
+            .map_err(Into::into)
+    }
+
+    /// Send `WaitSyncedRequest`
+    async fn wait_synced(
+        &self,
+        request: WaitSyncedRequest,
+        _timeout: Duration,
+    ) -> Result<tonic::Response<WaitSyncedResponse>, CurpError> {
+        let mut req = tonic::Request::new(request);
+        req.metadata_mut().inject_current();
+        self.server.wait_synced(req).await.map_err(Into::into)
+    }
+
+    /// Send `ShutdownRequest`
+    async fn shutdown(
+        &self,
+        request: ShutdownRequest,
+        _timeout: Duration,
+    ) -> Result<tonic::Response<ShutdownResponse>, CurpError> {
+        let mut req = tonic::Request::new(request);
+        req.metadata_mut().inject_current();
+        self.server.shutdown(req).await.map_err(Into::into)
+    }
+
+    /// Send `FetchClusterRequest`
+    async fn fetch_cluster(
+        &self,
+        request: FetchClusterRequest,
+        _timeout: Duration,
+    ) -> Result<tonic::Response<FetchClusterResponse>, CurpError> {
+        let mut req = tonic::Request::new(request);
+        req.metadata_mut().inject_current();
+        self.server.fetch_cluster(req).await.map_err(Into::into)
+    }
+
+    /// Send `FetchReadStateRequest`
+    async fn fetch_read_state(
+        &self,
+        request: FetchReadStateRequest,
+        _timeout: Duration,
+    ) -> Result<tonic::Response<FetchReadStateResponse>, CurpError> {
+        let mut req = tonic::Request::new(request);
+        req.metadata_mut().inject_current();
+        self.server.fetch_read_state(req).await.map_err(Into::into)
+    }
```

Comment:
Use macros to reduce repetition

Reply:
This file doesn't seem to be very large. I think it can be ignored.
If there are any changes later on (such as adding metrics), using macros is not suitable.